anvio/taxoestimation.py

```diff
@@ -0,0 +1,1201 @@
+        while len(newmatrix) > 2:
+            statistic, pvalue = ss.friedmanchisquare(*newmatrix)
+            print("statistic:%s pvalue:%f" % (statistic, pvalue))
+            if pvalue >= 0.05 or pvalue == "nan":
+                return(newmatrix, new_matrixlist_position_entry)
+                break
+            new_matrixlist_position_entry, newmatrix = self.del_higher_ranked_individu_matrix(
+                newmatrix, new_matrixlist_position_entry)
+        return(newmatrix, new_matrixlist_position_entry)
+
+    def rank_assignement(self, hits_per_gene, name):
+
+        matrix, matrix_pident, matrixlist_position_entry, list_position_ribosomal = self.make_rank_matrix(
+            name, hits_per_gene)
+        start_make_list_taxonomy = time.perf_counter()
+
+        taxonomy,bestlist = self.make_list_taxonomy(
+            matrix_pident, matrix, matrixlist_position_entry,list_position_ribosomal)
+
+        assignation_reduce = self.reduce_assignation_level(taxonomy,bestlist)
+        assignation = self.assign_taxonomie_solo_hit(assignation_reduce)
+
+        return(assignation)
+
+        end_make_list_taxonomy = time.perf_counter()
+        print("\n time make_list_taxonomy : ",
+              end_make_list_taxonomy - start_make_list_taxonomy)
+
+        """posthoc_conover=sp.posthoc_conover_friedman(matrix)
+        print("posthoc_conover")
+        print(posthoc_conover)
+        posthoc_miller_friedman=sp.posthoc_miller_friedman(matrix)
+        print("posthoc_miller_friedman")
+        print(posthoc_miller_friedman)
+        posthoc_siegel_friedman=sp.posthoc_siegel_friedman(matrix)
+        print("posthoc_siegel_friedman")
+        print(posthoc_siegel_friedman)
+        posthoc_durbin=sp.posthoc_durbin(matrix)
+        print("posthoc_durbin")
+        print(posthoc_durbin)
+
+        posthoc_nemenyi_friedman=sp.posthoc_nemenyi_friedman(matrix)
+        print("posthoc_nemenyi_friedman")
+        print(posthoc_nemenyi_friedman)"""
+
+    def del_higher_ranked_individu_matrix(self, matrix, matrixlist_position_entry):
+        maxrank = 0
+        i = 0
+        for liste in matrix:
+            summist = sum(list(liste))
+            if summist > maxrank:
+                maxrank = summist
+                dellist = i
+            i += 1
+        del matrix[dellist]
+        del matrixlist_position_entry[dellist]
+        return matrixlist_position_entry, matrix
+
+    def assign_taxonomie_solo_hit(self, taxonomy):
+        if not taxonomy or not taxonomy[0]:
+            return 0
+        assignation = taxonomy[0]
+
+        for s in taxonomy[1:]:
+            for key in s:
+                if s[key] not in assignation.values():
+                    if anvio.DEBUG:
+                        self.run.info('CONSIDER taxonomy','/'.join(assignation.values()))
+                    assignation.pop(key, None)
+        return(assignation)
+
+    def make_dicoidentitielevel(self):
```

Comment:
this function seems to be unnecessary for multiple reasons. one, it has all the hardcoded values and simply returns a dictionary. but also it does not seem to be called from anywhere. are there more of these orphan functions in the code that are not used by anyone? if yes, please remove them all :)